tests/tape/tapes/test_qnode.py

```diff
@@ -600,3 +601,40 @@ def func(x, y):
 
         assert np.allclose(res, res2, atol=tol, rtol=0)
         assert func.qtape is not old_tape
+
+
+class TestQNodeCollection:
+    """Unittests for the QNodeCollection"""
+
+    @pytest.fixture
+    def enable_tape_mode(self):
+        qml.enable_tape()
+        yield
+        qml.disable_tape()
+
+    def test_multi_thread(self, enable_tape_mode):
+        n_qubits = 4
+        n_batches = 5
+        dev = qml.device("default.qubit", wires=n_qubits)
+
+        qml.enable_tape()
+
+        def circuit(inputs, weights):
+            for index, input in enumerate(inputs):
+                qml.RY(input, wires=index)
+            for index in range(n_qubits - 1):
+                qml.CNOT(wires=(index, index + 1))
+            for index, weight in enumerate(weights):
+                qml.RX(weight, wires=index)
+            return [qml.expval(qml.PauliZ(wires=i)) for i in range(n_qubits)]
+
+        weight_shapes = {"weights": (n_qubits)}
+
+        qnode = QNodeCollection([QNode(circuit, dev) for _ in range(n_batches)])
+        x = np.random.rand(n_qubits).astype(np.float64)
+        p = np.random.rand(weight_shapes["weights"]).astype(np.float64)
+        try:
+            for _ in range(10):
+                qnode(x, p, parallel=True)

```

Comment:
Great test!

I'm wondering if there is a way to assert that the QNodes were created properly?

Reply:
Thanks! How about `try` / `except`ing arond the instantiation of the `QNodeCollection`?

Reply:
Can you remind me what the original error you were receiving before the thread locking was?

Reply:
It was like

```
 File "/home/shukob/anaconda3/envs/***/lib/python3.7/site-packages/PennyLane-0.13.0.dev0-py3.7.egg/pennylane/collections/qnode_collection.py", line 271, in __call__
    results = self.evaluate(args, kwargs)
  File "/home/shukob/anaconda3/envs/***/lib/python3.7/site-packages/PennyLane-0.13.0.dev0-py3.7.egg/pennylane/collections/qnode_collection.py", line 229, in evaluate
    return dask.compute(*results, scheduler=_scheduler)
  File "/home/shukob/anaconda3/envs/***/lib/python3.7/site-packages/dask/base.py", line 452, in compute
    results = schedule(dsk, keys, **kwargs)
  File "/home/shukob/anaconda3/envs/***/lib/python3.7/site-packages/dask/threaded.py", line 84, in get
    **kwargs
  File "/home/shukob/anaconda3/envs/***/lib/python3.7/site-packages/dask/local.py", line 486, in get_async
    raise_exception(exc, tb)
  File "/home/shukob/anaconda3/envs/***/lib/python3.7/site-packages/dask/local.py", line 316, in reraise
    raise exc
  File "/home/shukob/anaconda3/envs/***/lib/python3.7/site-packages/dask/local.py", line 222, in execute_task
    result = _execute_task(task, data)
  File "/home/shukob/anaconda3/envs/***/lib/python3.7/site-packages/dask/core.py", line 121, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
  File "/home/shukob/anaconda3/envs/***/lib/python3.7/site-packages/dask/utils.py", line 29, in apply
    return func(*args, **kwargs)
  File "/home/shukob/anaconda3/envs/***/lib/python3.7/site-packages/PennyLane-0.13.0.dev0-py3.7.egg/pennylane/tape/qnode.py", line 433, in __call__
    self.construct(args, kwargs)
  File "/home/shukob/anaconda3/envs/***/lib/python3.7/site-packages/PennyLane-0.13.0.dev0-py3.7.egg/pennylane/tape/qnode.py", line 395, in construct
    "All measurements must be returned in the order they are measured."
pennylane.qnodes.base.QuantumFunctionError: All measurements must be returned in the order they are measured.
```

Reply:
Ah, thanks! In that case, I think the way you have written the test currently works well - we are simply checking that the QNodes are constructed and called without an error occurring. 🙂